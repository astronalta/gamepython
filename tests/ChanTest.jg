# jgtest exec on

unbuffered() {
    ch = Chan[String](0)

    reader = Coroutine() func() {
        for event in ch {
            Io::println(event)
        }
    }
    reader()

    Io::println('put')
    ch.put('foo')
    Io::println('put')
    ch.put('bar')
    ch.close()
}

buffered() {
    ch = Chan[String](2)

    reader = Coroutine() func() {
        for event in ch {
            Io::println(event)
        }
    }

    Io::println('put')
    ch.put('foo')
    Io::println('put')
    ch.put('bar')
    ch.close()
    Io::manager().task(reader)
}

main() Int {
    unbuffered()
    buffered()
    ret 0
}
