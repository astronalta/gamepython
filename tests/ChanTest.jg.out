put
foo
put
bar
put
put
foo
bar
